sensor/common/scan/scan.go

```diff
@@ -3,62 +3,82 @@ package scan
 import (
 	"context"
 	"fmt"
+	"time"
 
 	"github.com/pkg/errors"
 	v1 "github.com/stackrox/rox/generated/api/v1"
 	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/errorhelpers"
 	"github.com/stackrox/rox/pkg/images/types"
-	"github.com/stackrox/rox/pkg/images/utils"
 	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/registries/docker"
+	"github.com/stackrox/rox/pkg/registries/rhel"
 	registryTypes "github.com/stackrox/rox/pkg/registries/types"
 	"github.com/stackrox/rox/pkg/signatures"
+	"github.com/stackrox/rox/pkg/urlfmt"
 	"github.com/stackrox/rox/sensor/common/registry"
 	"github.com/stackrox/rox/sensor/common/scannerclient"
 	scannerV1 "github.com/stackrox/scanner/generated/scanner/api/v1"
+	"golang.org/x/sync/semaphore"
+)
+
+const (
+	defaultMaxSemaphoreWaitTime = 5 * time.Second
 )
 
 var (
 	// ErrNoLocalScanner indicates there is no Secured Cluster local Scanner connection.
 	// This happens if it's not desired or if there is a connection error.
 	ErrNoLocalScanner = errors.New("No local Scanner connection")
 
+	// ErrTooManyParallelScans indicates there are too many scans in progress and wait time
+	// has been exceeded
+	ErrTooManyParallelScans = errors.New("too many parallel scans to local scanner")
+
+	redhatRegistryEndpoint = "registry.redhat.io"
+
 	log = logging.LoggerForModule()
 )
 
-// LocalScan wraps the functions required in EnrichLocalImage. This allows us to inject different values for testing purposes
+// LocalScan wraps the functions required for enriching local images. This allows us to inject different values for testing purposes
 type LocalScan struct {
-	// Used for testing purposes only to not require setting up registry / scanner.
```

Comment:
how come this was deleted?

Reply:
I felt it was misleading because its not 'only' used for testing purposes, its used during normal operation as well. The note seems sufficient to indicate tests are impacted by the assignment.